packages/flutter_tools/lib/src/web/file_generators/wasm_bootstrap.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+String generateWasmBootstrapFile() {
+  return r'''
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+(async function () {
+  let dart2wasm_runtime;
+  let moduleInstance;
+  try {
+    const dartModulePromise = WebAssembly.compileStreaming(fetch("main.dart.wasm"));
+    dart2wasm_runtime = await import('./dart2wasm_runtime.mjs');
+    moduleInstance = await dart2wasm_runtime.instantiate(dartModulePromise, {});
```

Comment:
I'm curious what's the reason for not doing compilation and instantiation together? Also, why not parallelize both `await`s:
```suggestion
    await Future.wait([
      import('./dart2wasm_runtime.mjs').then((value) {
        dart2wasm_runtime = value;
      }),
      WebAssembly.instantiateStreaming(fetch("main.dart.wasm"), {}).then((value) {
        moduleInstance = value;
      }),
    ]);
```

Reply:
`dart2wasm_runtime` is the thing that actually instantiates the module. It actually inserts a bunch of imports that are needed for the dart2wasm'd code to run properly (all the concrete implementations of JS interop stuff for example).

I'm pretty sure the code as-is is as concurrent as it possibly can be. `WebAssembly.compileStreaming(fetch("main.dart.wasm"))` returns a promise, so it's running concurrently with importing `dart2wasm_runtime.js`. But we must actually await `dart2wasm_runtime` for the next step which is instantiation.

Reply:
Oh oh.. I totally misread your code. Everything makes sense now.